extensions/panache/hibernate-orm-panache-kotlin/runtime/pom.xml

```diff
@@ -184,4 +173,35 @@
             </plugin>
         </plugins>
     </build>
+    <profiles>
+        <!-- Execute kapt, but not on Java 17 or higher due to https://youtrack.jetbrains.com/issue/KT-47583 -->
+        <profile>
+            <id>kapt-not-on-jdk17</id>
+            <activation>
+                <jdk>(,17)</jdk>
```

Comment:
does this work on 16 because of the --all-open bits?  what makes 17 fail so differently?

Reply:
A JDK class changed in 17, see https://youtrack.jetbrains.com/issue/KT-47583